### ISSUE TITLE
Scanner fails when provided OpenAPI specification file is of moderate size

### ISSUE THREAD

**Issue by svghadi** [RedHatProductSecurity/rapidast#175]
I am trying to run scans for Argo CD project. The OpenAPI specification file of Argo CD is not too huge, it is around 284KB. When I start the scan, it fails with below error:
```bash
Job openapi started
Job openapi target: https://localhost:8000/ invalid API URL: https://localhost:8000/swagger.json
Job openapi finished, time taken: 00:00:01
Automation plan failures:
        Job openapi target: https://localhost:8000/ invalid API URL: https://localhost:8000/swagger.json
WARNING:The ZAP process did not finish correctly, and exited with code 1
```

In the **zap.log**, I see some deserialization errors
```bash
2024-02-29 19:35:24,526 [main ] INFO  CommandLine - Job openapi started
2024-02-29 19:35:25,576 [main ] WARN  DeserializationUtils - Error snake-parsing yaml content
io.swagger.v3.parser.util.DeserializationUtils$SnakeException: Exception safe-checking yaml content  (maxDepth 2000, maxYamlAliasesForCollections 2147483647)
	at io.swagger.v3.parser.util.DeserializationUtils$CustomSnakeYamlConstructor.getSingleData(DeserializationUtils.java:483) ~[openapi-beta-37.zap:?]
	at org.yaml.snakeyaml.Yaml.loadFromReader(Yaml.java:493) ~[openapi-beta-37.zap:?]
.........

2024-02-29 19:35:25,639 [main ] ERROR DeserializationUtils - Error parsing content
com.fasterxml.jackson.dataformat.yaml.JacksonYAMLParseException: The incoming YAML document exceeds the limit: 3145728 code points.
 at [Source: (StringReader); line: 49813, column: 50]
..........

2024-02-29 19:35:25,702 [main ] WARN  OpenAPIV3Parser - Exception while parsing:
com.fasterxml.jackson.dataformat.yaml.JacksonYAMLParseException: The incoming YAML document exceeds the limit: 3145728 code points.
 at [Source: (StringReader); line: 49813, column: 50]
```

Looks like it is related to https://github.com/swagger-api/swagger-parser/pull/1872. I could workaround the failure by adding a global JAVA system property to increase the `maxYamlCodePoints` as suggested in the swagger-parser PR.
```bash
export _JAVA_OPTIONS=-DmaxYamlCodePoints=99999999
```

After adding `_JAVA_OPTIONS` I am able to run the scans successfully via Rapidast. 

### Enviroment
I am running Rapidast on Mac M1.

**Comment by cedricbu** [comment]
Thanks for the info & analysis !
We will try to find a nice way to integrate this. It's not the first time we see problems with large OpenAPI.
Maybe we could add a `misc` option such as `expect_large_openapi: [true|false]`

**Comment by jeremychoi** [comment]
> Maybe we could add a misc option such as expect_large_openapi: [true|false] 

It might be challenging to define 'large'. How about just adding one for configuring "maxYamlCodePoints=NNN"?

**Comment by cedricbu** [comment]
I created a PR for this: we have currently decided to simply add a troubleshooting note for adding the corresponding environment variable.
Also, a quick note: from my previous analysis of another instance of this particular crash, we found out that converting from Swagger v2 to Swagger v3 could also resolve the problem (at least in that particular case). So if you're using a Swagger v2 as source, it might be worth checking if updating to v3 helps.
